glean-core/android/src/main/java/mozilla/telemetry/glean/private/DatetimeMetricType.kt

```diff
@@ -59,9 +60,10 @@ class DatetimeMetricType internal constructor(
     }
 
     private fun shouldRecord(): Boolean {
-        // Don't record metrics if we aren't initialized
+        // If not initialized we cannot query across the FFI, so rely on getUploadEnabled() and the
+        // specific metric's enabled flag to determine shouldRecord's response
         if (!Glean.isInitialized()) {
```

Comment:
Why can't we simply remove this check?

Reply:
Because if Glean is not initialized in glean core, then the handle is `0` and the following call across the FFI goes kaboom

Reply:
Yes, I mean, I don't understand why we need to check this still. At this point, we don't really care if Glean is initialized or not anymore (on the Kotlin side). We could maybe just keep the check for the "should record" on the Rust part?

I'm suggesting to remove the Kotlin `shouldRecord` completely and only check on the Rust side, when we accumulate. Thoughts?

Reply:
I think we had thought about that, but decided it might be better to not queue any work on the dispatcher when shouldRecord is false (just to avoid the overhead of the dispatch).  But I don't recall actually measuring or confirming that.  Maybe it doesn't matter.

Reply:
Oh, okay that makes more sense.  The only reason I would say no is that it would allow us to not queue tasks if the Kotlin side were aware of this state, but with the size limitation of the queue, this may not be a big deal anyway.

Reply:
But the current `shouldRecord` checks for Glean to be initialized, which won't be a thing anymore. It *will* need to rely on the Dispatchers. After this patch, we could probably just not dispatch for metrics that are `disabled = true` (something that @travis79 already did as part of this patch).

Reply:
@badboy thoughts?

Reply:
* On the Rust side we (should) always check if we can record anyway
* So far we kept the shouldRecord checks on the Kotlin side to be consistent with glean-ac
    * One side-effect is that we don't dispatch if it is disabled anyway
    * We never measured if that has a performance impact.
* (Side-note: We don't crash on an invalid Glean handle in production, we do crash in debug mode. We _could_ change that if we need to)
* I think I do like @Dexterp37's idea: let's only check `disabled == true` on the Kotlin side, otherwise dispatch. The Rust side will take care of the additional check.
    * We should figure out how to measure if that eventually becomes to much dispatching and can change it at a later point?

Reply:
>       * We should figure out how to measure if that eventually becomes to much dispatching and can change it at a later point?

Yup, good idea. I'd be for filing a bug about that and take it for future investigation.

Reply:
Sounds good to me, I'll make the modifications to remove all but the checks for the metric `disabled == true`

Reply:
Bug filed for follow up investigation [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1569593).